src/app/api/categories/[categoryId]/route.ts

```diff
@@ -0,0 +1,22 @@
+import { response200, response404, response500 } from "@/app/utils";
+import { getProductsByCategory } from "@/model/ProductModel";
+
+export async function GET(
+  request: Request,
+  {
+    params,
+  }: {
+    params: { categoryId: string };
+  }
+) {
+  try {
+    let products = await getProductsByCategory(params.categoryId);
```

Comment:
What if the category exists BUT we have no products for it. It will return 404.

Here you should fetch the category first, then decide if it's 404 or if to continue.